### ISSUE TITLE
Update javapoet version

### ISSUE THREAD

**Issue by cogman** [google/auto#677]
When combining auto-value 1.6.2 is incompatible with dagger 2.19 due to autovalue having an older version of javapoet. (if autovalue appears first in the annotation processors path).

Updating autovalue's javapoet dependency to 1.11.1 would fix this.

Alternatively, it might be nice if javapoet was shaded by auto-value so these dependency mismatches don't matter.

Works
```
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-compiler-plugin</artifactId>
				<configuration>
					<annotationProcessorPaths>
						<path>
							<groupId>com.google.dagger</groupId>
							<artifactId>dagger-compiler</artifactId>
							<version>${dagger.version}</version>
						</path>
						<path>
							<groupId>com.google.auto.value</groupId>
							<artifactId>auto-value</artifactId>
							<version>${autovalue.version}</version>
						</path>
					</annotationProcessorPaths>
				</configuration>
```

Broken

```
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-compiler-plugin</artifactId>
				<configuration>
					<annotationProcessorPaths>
						<path>
							<groupId>com.google.auto.value</groupId>
							<artifactId>auto-value</artifactId>
							<version>${autovalue.version}</version>
						</path>
						<path>
							<groupId>com.google.dagger</groupId>
							<artifactId>dagger-compiler</artifactId>
							<version>${dagger.version}</version>
						</path>
					</annotationProcessorPaths>
				</configuration>
```

**Comment by tbroyer** [comment]
To me, the real, underlying issue is that Maven doesn't resolve to the greatest javapoet version, and prefers to downgrade Dagger than upgrade auto-value…

**Comment by cogman** [comment]
I don't disagree.

It is also a little annoying that, AFAIK, there isn't a way for me to explicitly set the java poet version for the two libraries.

Either way, though, it would be easier to update auto value than maven since the maven version selection process is basically set in stone.

**Comment by ronshapiro** [comment]
You can force the version by explicitly declaring the dependency version.

**Comment by bentatham** [comment]
I have the same issue.
As a workaround, rather than explicitly declaring the javapoet version in my pom, which might hinder future releases of dagger, I opted to add javapoet as an exclusion on auto-factory dependency, to force maven to use the transitive dependency version defined via dagger instead.

Would be much better if auto-factory was released though, and the beta dropped at some point soon.
